gammapy/modeling/models/spectral.py

```diff
@@ -1487,6 +1514,105 @@ def from_region_map(cls, map, **kwargs):
         return cls(energy=energy, values=values, **kwargs)
 
 
+class TemplateNDSpectralModel(SpectralModel):
+    """A model generated from a ND array where extra dimensions define the parameter space.
+
+    For more information see :ref:`templateND-spectral-model`.
+
+    Parameters
+    ----------
+    map : `~gammapy.maps.RegionNDMap`
+        Map template. 
+    meta : dict, optional
+        Meta information, meta['filename'] will be used for serialization
+    interp_kwargs : dict
+        Interpolation keyword arguments passed to `gammapy.maps.Map.interp_by_coord`.
+        Default arguments are {'interp': 'linear', 'fill_value': 0}.
+    """
+
+    tag = ["TemplateNDSpectralModel", "templateND"]
+
+    def __init__(self, map, interp_kwargs=None, meta=None, filename=None):
+        self._map = map.copy()
+        self.meta = dict() if meta is None else meta
+        if filename is not None:
+            filename = str(make_path(filename))
+        self.filename = filename
+
+        points_scale = [
+            "lin",
+            "lin",
+            "log",
+        ]
+        parameters = []
+        for axe in map.geom.axes:
+            if axe.name not in ["energy_true", "energy"]:
+                center = (axe.bounds[1] + axe.bounds[0]) / 2
+                parameter = Parameter(
+                    name=axe.name,
+                    value=center,
+                    unit=axe.unit,
+                    scale_method="scale10",
+                    min=axe.bounds[0],
+                    max=axe.bounds[-1],
+                    interp="lin",
+                )
+                points_scale.append("lin")
+                parameters.append(parameter)
+        self.default_parameters = Parameters(parameters)
+
+        interp_kwargs = interp_kwargs or {}
+        interp_kwargs.setdefault("values_scale", "log")
+        interp_kwargs.setdefault("points_scale", points_scale)
+        self._interp_kwargs = interp_kwargs
+        super().__init__()
+
+    @property
+    def map(self):
+        """Template map  (`~gammapy.maps.RegionNDMap`)"""
+        return self._map
+
+    def evaluate(self, energy, **kwargs):
+        lon = self.map.geom.center_skydir.data.lon.deg
+        lat = self.map.geom.center_skydir.data.lat.deg
+        coord = {"lon": lon, "lat": lat, "energy_true": energy}
+        coord.update(kwargs)
+
+        pixels = list(self.map.geom.coord_to_pix(coord))
+        # TODO : pixels in lon/lat are nan at center coord, why ?
```

Comment:
The usual behavior is that the lon / lat coordinate are `NaN` if the position is not contained in the region...this should be the case for many regions, but it's not guaranteed for compound regions or more complex shapes such as annuli or similar.